src/Tests/dotnet-workload-install.Tests/GivenDotnetWorkloadInstall.cs

```diff
@@ -368,6 +368,31 @@ public void GivenWorkloadInstallItErrorsOnInvalidWorkloadRollbackFile()
             string.Join(" ", _reporter.Lines).Should().Contain("Invalid rollback definition");
         }
 
+        [Fact]
+        public void GivenWorkloadInstallItErrorsWhenTheWorkloadIsAlreadyInstalled()
+        {
+            var testDirectory = _testAssetsManager.CreateTestDirectory().Path;
+            var dotnetRoot = Path.Combine(testDirectory, "dotnet");
+            var userProfileDir = Path.Combine(testDirectory, "user-profile");
+            var tmpDir = Path.Combine(testDirectory, "tmp");
+            var manifestPath = Path.Combine(_testAssetsManager.GetAndValidateTestProjectDirectory("SampleManifest"), "MockWorkloadsSample.json");
+            var workloadResolver = WorkloadResolver.CreateForTests(new MockManifestProvider(new[] { manifestPath }), dotnetRoot, false, userProfileDir);
+            var manifestUpdater = new MockWorkloadManifestUpdater();
+            var sdkFeatureVersion = "6.0.100";
+            var workloadId = "mock-1";
+
+            // Successfully install a workload
+            var installParseResult = Parser.Instance.Parse(new string[] { "dotnet", "workload", "install", workloadId });
+            var installCommand = new WorkloadInstallCommand(installParseResult, reporter: _reporter, workloadResolver: workloadResolver, nugetPackageDownloader: new MockNuGetPackageDownloader(tmpDir),
+                workloadManifestUpdater: manifestUpdater, userProfileDir: userProfileDir, version: sdkFeatureVersion, dotnetDir: dotnetRoot, tempDirPath: testDirectory);
+            installCommand.Execute();
+            _reporter.Clear();
+
+            // Install again, this time it should tell you that you already have the workload installed
+            installCommand.Execute();
```

Comment:
It might be a good idea for the second install command to specify a workload that was already installed, and one that should be installed, and to verify that the second one gets installed.